transformers/component_tests/queries/urn_bid_events_test.go

```diff
@@ -0,0 +1,85 @@
+	"github.com/makerdao/vdb-mcd-transformers/transformers/test_data"
+	"github.com/makerdao/vulcanizedb/libraries/shared/storage/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("urn bid events query", func() {
+	var (
+		usrOne, usrTwo     string
+		bidOneID, bidTwoID int
+		flipAddress        = test_data.EthFlipAddress()
+	)
+
+	BeforeEach(func() {
+		test_config.CleanTestDB(db)
+		timestamp := int64(rand.Int31())
+		blockNumber := rand.Int()
+		headerOneID := storageHelpers.CreateHeader(timestamp, blockNumber, db).Id
+		headerTwoID := storageHelpers.CreateHeader(timestamp+1, blockNumber+1, db).Id
+
+		bidOneID = rand.Int()
+		bidTwoID = rand.Int()
+		flipIlkMetadata := types.GetValueMetadata(storage.Ilk, nil, types.Bytes32)
+		bidKeys := map[types.Key]string{constants.BidId: strconv.Itoa(bidOneID)}
+		flipBidUsrMetadata := types.GetValueMetadata(storage.BidUsr, bidKeys, types.Address)
+
+		diffID := storageHelpers.CreateFakeDiffRecord(db)
+		usrOne = "0x" + test_data.RandomString(40)
+		usrTwo = "0x" + test_data.RandomString(40)
+
+		repo := flip.FlipStorageRepository{ContractAddress: flipAddress}
+		repo.SetDB(db)
+
+		// insert records used in join
+		ethIlkErr := repo.Create(diffID, headerOneID, flipIlkMetadata, test_helpers.FakeIlk.Hex)
+		Expect(ethIlkErr).NotTo(HaveOccurred())
+		ethBidUsrErr := repo.Create(diffID, headerOneID, flipBidUsrMetadata, usrOne)
+		Expect(ethBidUsrErr).NotTo(HaveOccurred())
+
+		flipBidUsrMetadata.Keys[constants.BidId] = strconv.Itoa(bidTwoID)
+		bidUsrErr := repo.Create(diffID, headerOneID, flipBidUsrMetadata, usrTwo)
+		Expect(bidUsrErr).NotTo(HaveOccurred())
+
+		// insert kicks for 2 different urns
+		logID := test_data.CreateTestLog(headerOneID, db).ID
+		kickErrOne := test_helpers.CreateFlipKick(flipAddress, bidOneID, headerOneID, logID, usrOne, db)
+		Expect(kickErrOne).NotTo(HaveOccurred())
+		kickErrTwo := test_helpers.CreateFlipKick(flipAddress, bidTwoID, headerTwoID, logID, usrTwo, db)
+		Expect(kickErrTwo).NotTo(HaveOccurred())
+	})
+
+	It("only gets bid events related to the passed-in urn", func() {
+		expectedFlip := []test_helpers.BidEvent{{ContractAddress: flipAddress, BidId: strconv.Itoa(bidOneID)}}
+
+		var actualEvents []test_helpers.BidEvent
+		err := db.Select(&actualEvents, `SELECT contract_address, bid_id  FROM api.urn_bid_events($1, $2)`,
+			usrOne, test_helpers.FakeIlk.Identifier)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(actualEvents).To(ConsistOf(expectedFlip))
+	})
+
+	It("gets another urn's bid events", func() {
+		expectedFlip := []test_helpers.BidEvent{{ContractAddress: flipAddress, BidId: strconv.Itoa(bidTwoID)}}
+
+		var actualEvents []test_helpers.BidEvent
+		err := db.Select(&actualEvents, `SELECT contract_address, bid_id  FROM api.urn_bid_events($1, $2)`,
+			usrTwo, test_helpers.FakeIlk.Identifier)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(actualEvents).To(ConsistOf(expectedFlip))
+	})
```

Comment:
Reluctant to suggest this because these tests already require some weighty setup, but I think it'd be good to add a row for a different ilk and verify that we're correctly filtering on that field

Reply:
One other test that might provide helpful documentation could mention that it provides results from multiple auctions for an urn (if that urn has been flipped several times)